src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -236,6 +236,23 @@ private void ValidateLinuxSKUConfiguration()
             {
                 _logger.LogError($"Unable to load the functions payload since the app was not provisioned with valid {AzureWebJobsSecretStorage} connection string.");
             }
+
+            var functionsTimeZone = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsTimeZone);
+            var functionsWebsiteTimeZone = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsWebsiteTimeZone);
+
+            // If we have a linux consumption app and the time zone env variable is set, log a warning and diagnostic event
+            if (EnvironmentExtensions.IsLinuxConsumptionOnAtlas(_environment) &&
+                (!string.IsNullOrEmpty(functionsTimeZone) ||
+                !string.IsNullOrEmpty(functionsWebsiteTimeZone)))
+            {
+                string message = Script.Properties.Resources.LinuxConsumptionRemoveTimeZone;
+
+                // Log diagnostic event
+                DiagnosticEventLoggerExtensions.LogDiagnosticEventError(_logger, DiagnosticEventConstants.LinuxConsumptionTimeZoneErrorCode, message, DiagnosticEventConstants.LinuxConsumptionTimeZoneErrorHelpLink, new Exception(message));
```

Comment:
Using `_logger` may not be best here (same with the call above, which was already here). We have 2 levels of loggers in Functions host:
- the WebHost-level one... which does not log to for customers App Insights
- the JobHost-level one... which does

There's a helper in this class -- `GetHostLogger()` which pulls the JobHost-level logger, if it exists, otherwise, falls back to the WebHost. Both will show up in Kusto, but we want to use the JobHost if at all possible.

All that being said, I'd propose:
- change this method to take in an `ILogger` as a parameter.
- change everything in here to use that instead of `_logger`
- when you call this, pass it the return value from `GetHostLogger(localhost)`

Reply:
Done! Let me know if this is correct